app/views/submissions/_publishing.html.haml

```diff
@@ -0,0 +1,6 @@
+- if submission.publishable? && submission.revocable?
+  = confirmable_action("Revoke publication", submission,
+      url: revoke_submission_path(submission), method: :patch)
+- elsif !submission.publishable?
```

Comment:
I still think that it should be `Submission#published?` and not `#publishable?`. Now it looks like the "Publish" button is shown for submissions which are not publishable :)